app/frontend/src/components/NotificationBar/NotificationBar.tsx

```diff
@@ -0,0 +1,56 @@
+// Eternal Imports
+import { combineClasses } from "components/Utility/utils";
+import React, { useState, useRef } from "react";
+import { CSSTransition } from "react-transition-group";
+
+// Internal Imports
+import { iconX } from "../../assets/images/images";
+// Type declaration for props
+interface NotificationBarProps extends React.PropsWithChildren {
+  autoHidden: boolean; // 1. Added new prop to indicate the bar disappearing on its own
+  color: String;
+  text: String;
+}
+
+function NotificationBar({
+  autoHidden = false,
+  color = "warning",
+  text = "Content of the notification appears here",
+  ...props
+}: NotificationBarProps) {
+  const [showBar, setShowBar] = useState(true);
+  const nodeRef = useRef(null);
+
+  // 2. Added CSS Transition component to wrap around notification bar
+  return (
+    <CSSTransition
+      in={showBar}
+      nodeRef={nodeRef}
+      classNames="notification-bar-fade"
+      timeout={300}
```

Comment:
This fade value might be too low, since the element blinks out immediately, making it hard to read the notification.